### PR TITLE
add log to gas deposit

### PR DESCRIPTION
**Pull request by CoderZhi**
Unit tests are not added yet.

### CODE REVIEW

File: action/protocol/staking/handlers.go [hunk 1]
```diff
@@ -649,40 +649,45 @@ func (p *Protocol) handleCandidateRegister(ctx context.Context, act *action.Cand
...
 	// put updated caller's account state to trie
 	if err := accountutil.StoreAccount(csm, actCtx.Caller, caller); err != nil {
-		return log, nil, nil, errors.Wrapf(err, "failed to store account %s", actCtx.Caller.String())
+		return log, nil, errors.Wrapf(err, "failed to store account %s", actCtx.Caller.String())
 	}
 
 	// put registrationFee to reward pool
-	if err := p.depositGas(ctx, csm, registrationFee); err != nil {
-		return log, nil, nil, errors.Wrap(err, "failed to deposit gas")
+	depositLog, err := p.depositGas(ctx, csm, registrationFee)
```
Comment (yutongp) [inline action/protocol/staking/handlers.go:676]: I think this is handled already on line 703

Comment (CoderZhi) [reply action/protocol/staking/handlers.go:676]: yes, let me think about a good way to resolve this problem.

File: action/protocol/execution/evm/evm.go [hunk 2]
```diff
@@ -193,10 +194,25 @@ func ExecuteContract(
 			remainingValue := new(big.Int).Mul(new(big.Int).SetUint64(remainingGas), ps.context.GasPrice)
 			stateDB.AddBalance(ps.context.Origin, remainingValue)
 		}
+		if depositGas-remainingGas > 0 {
+			burnLog = &action.Log{
+				Address:     contractAddress,
+				BlockHeight: blkCtx.BlockHeight,
+				ActionHash:  execution.Hash(),
```
Comment (dustinxie) [inline action/protocol/execution/evm/evm.go:201]: don't need these 3, won't be used

File: action/protocol/execution/evm/evm.go [hunk 3]
```diff
@@ -206,6 +222,12 @@ func ExecuteContract(
 	}
 	stateDB.clear()
 	receipt.Logs = stateDB.Logs()
```
Comment (dustinxie) [inline action/protocol/execution/evm/evm.go:224]: nit: should we switch order of these 2 lines?
and in clear(), also do statedb.Logs = make([]*action.Log)
maybe another PR

**Pull request merged**